src/unix.rs

```diff
@@ -155,6 +159,21 @@ impl Client {
             (read_err, write_err) => {
                 read_err?;
                 write_err?;
+
+                // Optimization: Try converting it to a fifo by using /dev/fd
+                //
+                // On linux, opening `/dev/fd/$fd` returns a fd with a new file description,
+                // so we can set `O_NONBLOCK` on it without affecting other processes.
+                //
+                // On macOS, opening `/dev/fd/$fd` seems to be the same as `File::try_clone`.
+                //
+                // I tested this on macOS 14 and Linux 6.5.13
+                #[cfg(target_os = "linux")]
+                if let Ok(Some(jobserver)) =
+                    Self::from_fifo(&format!("/dev/fd/{}", read.as_raw_fd()))
```

Comment:
This optimization doesn't look correct. The string need to be prefixed with `fifo:` in order to make it work.

Reply:
BTW, just found this post <https://utcc.utoronto.ca/~cks/space/blog/unix/DevFdAndDup> so it seems fine to just open `/dev/fs/N`.

Reply:
Thanks, I've added `fifo:` prefix to it.